src/Function/Construct/Symmetry.agda

```diff
@@ -0,0 +1,139 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some functional properties are symmetric
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Construct.Symmetry where
+
+open import Data.Product using (_,_; swap)
+open import Function
+open import Level using (Level)
+open import Relation.Binary hiding (_⇔_)
+
+private
+  variable
+    a b c ℓ₁ ℓ₂ ℓ₃ : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+-- Properties
+
+module _ (≈₁ : Rel A ℓ₁) (≈₂ : Rel B ℓ₂)
+         (f : A → B) {f⁻¹ : B → A}
+         where
+
+  inverseʳ : Inverseˡ ≈₁ ≈₂ f f⁻¹ → Inverseʳ ≈₂ ≈₁ f⁻¹ f
+  inverseʳ inv = inv
+
+  inverseˡ : Inverseʳ ≈₁ ≈₂ f f⁻¹ → Inverseˡ ≈₂ ≈₁ f⁻¹ f
+  inverseˡ inv = inv
+
+  inverseᵇ : Inverseᵇ ≈₁ ≈₂ f f⁻¹ → Inverseᵇ ≈₂ ≈₁ f⁻¹ f
+  inverseᵇ (invˡ , invʳ) = (invʳ , invˡ)
+
+------------------------------------------------------------------------
+-- Structures
+
+module _ {≈₁ : Rel A ℓ₁} {≈₂ : Rel B ℓ₂}
+         {f : A → B} {f⁻¹ : B → A}
+         where
+
+  isLeftInverse : IsRightInverse ≈₁ ≈₂ f f⁻¹ → IsLeftInverse ≈₂ ≈₁ f⁻¹ f
+  isLeftInverse inv = record
+    { isCongruent = record
+      { cong = F.cong₂
+      ; isEquivalence₁ = F.isEquivalence₂
+      ; isEquivalence₂ = F.isEquivalence₁
+      }
+    ; cong₂ = F.cong₁
+    ; inverseˡ = inverseˡ ≈₁ ≈₂ f {f⁻¹} F.inverseʳ
+    }
+    where module F = IsRightInverse inv
+
+  isRightInverse : IsLeftInverse ≈₁ ≈₂ f f⁻¹ → IsRightInverse ≈₂ ≈₁ f⁻¹ f
+  isRightInverse inv = record
+    { isCongruent = record
```

Comment:
Equally if you split `isCongruent` out then you could reuse it in the left/right inverse proofs.

Reply:
Done.